tests/queries/0_stateless/03660_udf_subquery.sql

```diff
@@ -0,0 +1,33 @@
+-- Tags: no-parallel
+
+SET enable_analyzer = 1;
+
+DROP TABLE IF EXISTS dict;
+create table dict engine=MergeTree() order by id as
+select 1 as id, 'one' as name union all
+select 2 as id, 'two' as name;
+
+CREATE OR REPLACE FUNCTION udf_type_of_int AS
+int_ -> (select if(name = 'one', 'The One', 'other') from dict where id = int_);
+
+-- this part worked successfully
+SELECT * FROM (
```

Comment:
This query doesn't work without the change in `src/Analyzer/Resolve/evaluateScalarSubqueryIfNeeded.cpp`